From aa5c2c7af69e3f733d92093a009c1e1dd510bb76 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Thu, 28 Sep 2017 15:38:49 +0800
Subject: [PATCH 2633/5242] MLK-16582 ARM64: dts: imx8qxp: mek: add uart port2
 and port3 for base board

commit  3e7ea4a063be6899ffc0962b0c9ed91439f6fcd7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add uart port2 and port3 for MEK base board.

BuildInfo:
 - SCFW 9e9f6ec6, IMX-MKIMAGE 0, ATF 0
 - U-Boot 2017.03-imx_v2017.03_4.9.11_imx8_alpha+g258936c

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |   38 +++++++++++++++++----
 1 file changed, 32 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index abe7a94..fd059ca 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -204,17 +204,31 @@
 
 		pinctrl_lpuart0: lpuart0grp {
 			fsl,pins = <
-				SC_P_UART0_RX_ADMA_UART0_RX	0x0600002c
-				SC_P_UART0_TX_ADMA_UART0_TX	0x0600002c
+				SC_P_UART0_RX_ADMA_UART0_RX	0x06000020
+				SC_P_UART0_TX_ADMA_UART0_TX	0x06000020
 			>;
 		};
 
 		pinctrl_lpuart1: lpuart1grp {
 			fsl,pins = <
-				SC_P_UART1_TX_ADMA_UART1_TX		0x0600002c
-				SC_P_UART1_RX_ADMA_UART1_RX		0x0600002c
-				SC_P_UART1_RTS_B_ADMA_UART1_RTS_B	0x0600002c
-				SC_P_UART1_CTS_B_ADMA_UART1_CTS_B	0x0600002c
+				SC_P_UART1_TX_ADMA_UART1_TX		0x06000020
+				SC_P_UART1_RX_ADMA_UART1_RX		0x06000020
+				SC_P_UART1_RTS_B_ADMA_UART1_RTS_B	0x06000020
+				SC_P_UART1_CTS_B_ADMA_UART1_CTS_B	0x06000020
+			>;
+		};
+
+		pinctrl_lpuart2: lpuart2grp {
+			fsl,pins = <
+				SC_P_UART2_TX_ADMA_UART2_TX	0x06000020
+				SC_P_UART2_RX_ADMA_UART2_RX	0x06000020
+			>;
+		};
+
+		pinctrl_lpuart3: lpuart3grp {
+			fsl,pins = <
+				SC_P_FLEXCAN2_TX_ADMA_UART3_TX	0x06000020
+				SC_P_FLEXCAN2_RX_ADMA_UART3_RX	0x06000020
 			>;
 		};
 
@@ -409,6 +423,18 @@
 	status = "okay";
 };
 
+&lpuart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart2>;
+	status = "okay";
+};
+
+&lpuart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart3>;
+	status = "okay";
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec1>;
-- 
1.7.9.5

